migrations/postgres/20240422100311_flow_configuration_events.sql

```diff
@@ -0,0 +1,26 @@
+CREATE SEQUENCE flow_configuration_event_id_seq AS BIGINT;
+
+CREATE TYPE system_flow_type AS ENUM ('gc');
+
+CREATE TABLE system_flow_configuration_events
+(
+    event_id         BIGINT PRIMARY KEY DEFAULT NEXTVAL('flow_configuration_event_id_seq'),
+    system_flow_type system_flow_type NOT NULL,
+    event_type       VARCHAR(50)      NOT NULL,
+    event_time       TIMESTAMPTZ      NOT NULL,
+    event_payload    JSONB            NOT NULL
+);
+
+CREATE TYPE dataset_flow_type AS ENUM ('ingest', 'execute_transform', 'hard_compacting');
```

Comment:
Let's think of an example (maybe put it in comment) how we would extend this list if needed. We will likely be adding new types of flows in nearest months for sure.

I remember this wasn't trivial in Postgres. If I'm not mistaken, this would require creating new type and temporary columns. Or there was a solution that needs admin permissions and accesses `pg_*` system tables.

Reply:
In general, adding isn't the problem -- more of a challenge to remove.

Added a couple comments -- Done

Reply:
How to rename?

Reply:
Added notes about that -- long story short: easy w/o any tricks

Done

Reply:
The existing unit-test has been updated to handle this kind of mistakes